src/test/datascience/editor-integration/hoverProvider.vscode.test.ts

```diff
@@ -0,0 +1,50 @@
+import { assert } from 'chai';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { IHoverProvider } from '../../../client/datascience/types';
+import { IExtensionTestApi, openFile, sleep } from '../../common';
+import { EXTENSION_ROOT_DIR_FOR_TESTS } from '../../constants';
+import { initialize } from '../../initialize';
+
+suite('Hover provider', async () => {
+    const file = path.join(
+        EXTENSION_ROOT_DIR_FOR_TESTS,
+        'src',
+        'test',
+        'datascience',
+        'editor-integration',
+        'tooltips.py'
+    );
+    let api: IExtensionTestApi;
+    suiteSetup(async function () {
+        if (process.env.VSC_FORCE_REAL_JUPYTER === undefined) {
+            return this.skip();
+        }
+        api = await initialize();
+    });
+    test('Tensor tooltips', async () => {
+        // Open a Python file
+        const textDocument = await openFile(file);
+
+        // Wait for code lenses to get detected.
+        await sleep(1_000);
+
+        // Execute its contents in the interactive window
+        await vscode.commands.executeCommand<void>('jupyter.runallcells', textDocument.uri);
+
+        // Request a hover on the line containing a tensor variable declaration
+        const hoverProvider = api.serviceContainer.get<IHoverProvider>(IHoverProvider);
+        const position = new vscode.Position(11, 1);
+        const cancelTokenSource = new vscode.CancellationTokenSource();
+        const result = await hoverProvider.provideHover(textDocument, position, cancelTokenSource.token);
```

Comment:
Test is still failing because kernelVariables doesn't appear to know anything about our variables, will fix before merge.